File: JDK11/opentraceability/src/main/java/opentraceability/models/masterdata/Address.java
package opentraceability.models.masterdata;

import java.util.ArrayList;
import java.util.List;

import opentraceability.models.common.LanguageString;
import opentraceability.utility.Country;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;
import opentraceability.utility.attributes.OpenTraceabilityMasterDataAttribute;

public class Address {
    @OpenTraceabilityJsonAttribute(name = "@type")
    public String type = "gs1:PostalAddress";

    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#streetAddressOne")
    @OpenTraceabilityJsonAttribute(name = "streetAddress")
    public List<LanguageString> address1 = new ArrayList<>();

    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#streetAddressTwo")
    public List<LanguageString> address2 = new ArrayList<>();

    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#city")
    @OpenTraceabilityJsonAttribute(name = "addressLocality")
    public List<LanguageString> city = new ArrayList<>();

    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#state")
    @OpenTraceabilityJsonAttribute(name = "addressRegion")
    public List<LanguageString> state = new ArrayList<>();

    @OpenTraceabilityMasterDataAttribute(name = "urn:epcglobal:cbv:mda#countryCode")
    @OpenTraceabilityJsonAttribute(name = "countryCode")
    public Country country;
}

File: Java/OpenTraceability/src/main/opentraceability/models/masterdata/address.kt
package models.masterdata

import models.common.LanguageString
import utility.Country
import utility.attributes.OpenTraceabilityJsonAttribute
import utility.attributes.OpenTraceabilityMasterDataAttribute
import java.util.*
import java.lang.reflect.Type

class Address {

    @OpenTraceabilityJsonAttribute("@type")
    var Type: String? = "gs1:PostalAddress"

    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#streetAddressOne")
    @OpenTraceabilityJsonAttribute("streetAddress")
    var Address1: ArrayList<LanguageString>? = null

    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#streetAddressTwo")
    var Address2: ArrayList<LanguageString>? = null

    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#city")
    @OpenTraceabilityJsonAttribute("addressLocality")
    var City: ArrayList<LanguageString>? = null

    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#state")
    @OpenTraceabilityJsonAttribute("addressRegion")
    var State: ArrayList<LanguageString>? = null

    @OpenTraceabilityMasterDataAttribute("urn:epcglobal:cbv:mda#countryCode")
    @OpenTraceabilityJsonAttribute("countyCode")
    var Country: Country? = null
}
